tests/selenium_tests.py

```diff
@@ -0,0 +1,122 @@
+                return location.find_element_by_link_text(location_name).click()
+        except NoSuchElementException:
+            message = 'There is no location with {} name'.format(location_name)
+            self.fail(message)
+
+    def set_location_marker(self, driver, location_name):
+        # set marker on map on location object
+        self.get_location(driver, location_name)
+        marker_button = driver.find_element_by_class_name('leaflet-draw-draw-marker')
+        marker_button.click()
+        geo_map = driver.find_element_by_id('id_geometry-map')
+        geo_map.click()
+        save_button = driver.find_element_by_name('_save')
+        save_button.click()
+
+    def delete_location(self, driver, location_name):
+        self.get_location(driver, location_name)
+        delete_button = (
+            driver.find_element_by_class_name('submit-row').
+            find_element_by_class_name('deletelink-box')
+        )
+        delete_button.click()
+        confirm_delete_button = (
+            driver.
+            find_element_by_xpath('//div/div[@id="content"]/form/div/input[@type="submit"]')
+        )
+        confirm_delete_button.click()
+
+
+class TestWebSocket(TestUtilities, unittest.TestCase):
+    is_headless = False
+
+    def setUp(self):
+        self.chrome_options = Options()
+        if self.is_headless:
+            self.chrome_options.add_argument("--headless")
+        self.base_driver = webdriver.Chrome(options=self.chrome_options)
+        self.location_name = 'automated-websocket-selenium-loc01'
+
+    def tearDown(self):
+        self.delete_location(self.base_driver, self.location_name)
+        self.base_driver.close()
+
+    def test_websocket(self):
+        self.log_in(self.base_driver)
+        self.create_mobile_location(self.base_driver, self.location_name)
+        self.get_location(self.base_driver, self.location_name)
+        js_websocket_errors = []
+        for error in self.get_error_js_logs(self.base_driver):
+            if 'websocket' or 'ws' in error.lower():
+                js_websocket_errors.append(error)
+        if js_websocket_errors:
+            self.fail('Following WebSocket related js errors occurred: ' + ', '.join(js_websocket_errors))
+        # assert that there is no markers on map
+        self.assertEqual(len(self.base_driver.find_elements_by_class_name('leaflet-marker-icon')), 0)
+        change_location_driver = webdriver.Chrome(options=self.chrome_options)
+        self.log_in(change_location_driver)
+        self.set_location_marker(change_location_driver, self.location_name)
+        change_location_driver.close()
+        # assert that marker popped up on map
+        self.assertEqual(len(self.base_driver.find_elements_by_class_name('leaflet-marker-icon')), 1)
+
+
+if __name__ == '__main__':
+    # To run tests in chrome headless mode pass --headless command line argument
+    if '--headless' in sys.argv:
+        sys.argv.remove('--headless')
```

Comment:
Don't remove it. 
Might come in handly somewhere else in future! :smile:

Reply:
If I won't delete it I get `error: unrecognized arguments: --headless` error, popping it also removes it

Reply:
Okay. I'll look into it later sometime! :smile:

Reply:
You are using `sys.argv` both `headless` and `--headless` do the same job, I think you wanted to use `argparser` in which `--headless` is a optional arguement with a possible value like `--headless True`.
In anycase, I tired argparser today, I can't work without a ugly hack here, so I dropped the idea and moved to a json file. :smile: